### PR TITLE
UT for OnInteriorVehicleData

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
UT for OnInteriorVehicleData

### CODE REVIEW

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+using ::application_manager::ApplicationSharedPtr;
+using ::protocol_handler::MessagePriority;
+using can_cooperation::CANModuleInterface;
+using can_cooperation::MessageHelper;
+using namespace can_cooperation;
+
+namespace {
+const int kModuleId = 153;
+
+const std::string json =
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra Variable named wrongly. Not according to project coding style

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  Json::Value module_type =
+      json_value[json_keys::kParams][message_params::kModuleData]
+                [message_params::kModuleType];
+  can_app_extention_->SubscribeToInteriorVehicleData(module_type);
+  apps_.push_back(mock_app_);
+  // Expectations
+  EXPECT_CALL(mock_module_, service()).WillOnce(Return(mock_service_));
+  EXPECT_CALL(*mock_service_, GetApplications(kModuleId))
+      .WillOnce(Return(apps_));
+  application_manager::MessagePtr result_msg;
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra 
Move it to the place before its first usage

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  EXPECT_CALL(*mock_app_, QueryInterface(kModuleId))
+      .WillOnce(Return(can_app_extention_));
+  ON_CALL(*mock_app_, app_id()).WillByDefault(Return(app_id_));
+  EXPECT_CALL(*mock_service_, GetApplication(app_id_))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_service_, CheckPolicyPermissions(_))
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra Parameter verification?

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      .WillOnce(Return(mobile_apis::Result::eType::SUCCESS));
+  EXPECT_CALL(*mock_service_, CheckModule(app_id_, "CLIMATE"))
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra Use constant instead of hardcoded value

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    return message;
+  }
+
+ protected:
+  utils::SharedPtr<NiceMock<application_manager::MockService> > mock_service_;
+  utils::SharedPtr<NiceMock<MockApplication> > mock_app_;
+  utils::SharedPtr<can_cooperation::CANAppExtension> can_app_extention_;
+  can_cooperation_test::MockCANModuleInterface mock_module_;
+  std::vector<ApplicationSharedPtr> apps_;
+  const uint32_t app_id_ = 11u;
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra Why did you make it class member?

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+TEST_F(OnInteriorVehicleDataNotificationTest, Execute_AppIsSubscribed) {
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra Test case name does not match its logic. Please rename.

File: src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  EXPECT_EQ(application_manager::MessageType::kNotification,
+            result_msg->type());
+  EXPECT_EQ(functional_modules::ON_INTERIOR_VEHICLE_DATA,
+            result_msg->function_id());
+  EXPECT_EQ(MessageHelper::GetMobileAPIName(
+                functional_modules::ON_INTERIOR_VEHICLE_DATA),
+            result_msg->function_name());
+}
+
+TEST_F(OnInteriorVehicleDataNotificationTest, Execute_AppIsUnsubscribed) {
```
Comment (AByzhynar) [inline src/components/can_cooperation/test/commands/on_interior_vehicle_data_notification_test.cc]: @LitvinenkoIra Test case name does not match its logic. Please rename.

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Review by LuxoftAKutsan — APPROVED** [review]

**Review by AByzhynar — APPROVED** [review]

**Pull request merged**